src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -583,26 +581,28 @@ def _update_buffer_len(self) -> bool:
         Returns:
             Whether the buffer length was changed (was really updated).
         """
-        input_sampling_period_s = self._source_properties.sampling_period_s
-
         # To make type checking happy
-        assert input_sampling_period_s is not None
         assert self._buffer.maxlen is not None
+        assert self._source_properties.sampling_period is not None
+
+        input_sampling_period = self._source_properties.sampling_period
 
         config = self._config
 
         # If we are upsampling, one sample could be enough for back-filling, but
         # we store max_data_age_in_periods for input periods, so resampling
         # functions can do more complex inter/extrapolation if they need to.
-        if input_sampling_period_s > config.resampling_period_s:
-            new_buffer_len = input_sampling_period_s * config.max_data_age_in_periods
+        if input_sampling_period > config.resampling_period:
+            new_buffer_len = (
+                input_sampling_period.total_seconds() * config.max_data_age_in_periods
+            )
         # If we are upsampling, we want a buffer that can hold
-        # max_data_age_in_periods * resampling_period_s seconds of data, and we
-        # one sample every input_sampling_period_s.
+        # max_data_age_in_periods * resampling_period of data, and we
+        # one sample every input_sampling_period.
         else:
             new_buffer_len = (
-                config.resampling_period_s
-                / input_sampling_period_s
+                config.resampling_period.total_seconds()
+                / input_sampling_period.total_seconds()
                 * config.max_data_age_in_periods
             )
```

Comment:
not entirely related to this PR, but I'd like to mention it anyway :) I think this is not a good practice in the sense that here `new_buffer_len` is defined dynamically as a `float` where in other places above it is an int. So we should probably use `math.ceil()` here instead

Reply:
Yeah, the `ceil` is done afterwards:

```py
        else:
            new_buffer_len = (
                config.resampling_period.total_seconds()
                / input_sampling_period.total_seconds()
                * config.max_data_age_in_periods
            )

        new_buffer_len = max(1, math.ceil(new_buffer_len))
#                               ^^^^^^^^^^^^^^^^^^^^^^^^^
```

It is weird though that the `new_buffer_len` is a `float`, so I will add a commit to improve the typing.